rerun_cpp/src/rerun/component_type_registry.cpp

```diff
@@ -0,0 +1,27 @@
+#include "component_type_registry.hpp"
+
+namespace rerun {
+
+    /// Returns the handle to the registered component type for the given descriptor.
+    ///
+    /// Registers the component type when first encountered.
+    Result<ComponentTypeHandle> ComponentTypeRegistry::get_or_register(
+        const ComponentDescriptor& descriptor, std::shared_ptr<arrow::DataType> arrow_datatype
```

Comment:
nit:

```suggestion
        const ComponentDescriptor& descriptor, const std::shared_ptr<arrow::DataType>& arrow_datatype
```

That way we won't needlessly bump the ref count if we don't have to